File: c 풀이/C 프로그래밍 2 D0.c
#include <stdio.h>

int m;
char a[100+10][100+10];
int p;
char pattern[100 + 10][100 + 10];
int ans;

void input(void)
{
	scanf("%d", &m);
	for (int i = 0; i < m; i++) {
		scanf(" %s", a[i]);
	}
	scanf("%d", &p);
	for (int i = 0; i < p; i++) {
		scanf(" %s", pattern[i]);
	}
}

int is_pattern(int x, int y)
{
	for (int i = x; i < x + p; i++) {
		for (int j = y; j < y + p; j++) {
			if (a[i][j] != pattern[i - x][j - y]) return 0;
		}
	}
	return 1;
}

void func(void)
{
	for (int i = 0; i <= m - p; i++) {
		for (int j = 0; j <= m - p; j++) {
			if (is_pattern(i, j)) ans += 1;
		}
	}
}

void main(void)
{
	input();
	func();
	printf("%d\n", ans);
}

File: python 풀이/C 프로그래밍 2 D0.py
m = int(input())
arr = [list(input()) for _ in range(m)]
n = int(input())
pattern = [list(input()) for _ in range(n)]

def find_pattern(x, y) :
    for i in range(n) :
        for j in range(n) :
            if arr[x+i][y+j] != pattern[i][j] : return 0
    return 1

ans = 0
for i in range(m-n+1) :
    for j in range(m-n+1) :
        ans += find_pattern(i, j)
print(ans)